staging/src/k8s.io/api/admissionregistration/v1beta1/types.go

```diff
@@ -259,8 +259,33 @@ type Webhook struct {
 	// Default to `['v1beta1']`.
 	// +optional
 	AdmissionReviewVersions []string `json:"admissionReviewVersions,omitempty" protobuf:"bytes,8,rep,name=admissionReviewVersions"`
+
+	// reinvocationPolicy indicates whether this webhook should be called multiple times as part of a single admission evaluation.
+	// Allowed values are "Never" and "IfNeeded".
+	//
+	// Never: the webhook will not be called more than once in a single admission evaluation.
+	//
+	// IfNeeded: the webhook will be called at least one additional time as part of the admission evaluation
+	// if the object being admitted is modified by other admission plugins after the initial webhook call.
+	// Webhooks that specify this option *must* be idempotent, able to process objects they previously admitted.
+	// Note:
+	// * the number of additional invocations is not guaranteed to be exactly one.
+	// * if additional invocations result in further modifications to the object, webhooks are not guaranteed to be invoked again.
+	// * webhooks that use this option may be reordered to minimize the number of additional invocations.
+	// * to validate an object after all mutations are guaranteed complete, use a validating admission webhook instead.
+	//
+	// Defaults to "Never".
+	// +optional
+	ReinvocationPolicy *ReinvocationPolicyType `json:"reinvocationPolicy,omitempty" protobuf:"bytes,9,opt,name=reinvocationPolicy,casttype=ReinvocationPolicyType"`
```

Comment:
what's the rational for a pointer?

Reply:
all optional fields are asked to be pointers (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required)

Reply:
Then we are not very consequent with optional strings in existing types. They a non-pointers everywhere.

Reply:
Looks like within the `Webhook` type we also use a pointer for the `FailurePolicy` and `SideEffects` string fields. So for consistency with the other fields in the type, and to adhere with documented API conversion I'll keep it as a pointer.